app/models/manageiq/providers/kubevirt/infra_manager.rb

```diff
@@ -122,13 +123,20 @@ def self.verify_credentials(args)
   # @option opts [String] :token The Kubernetes authentication token.
   #
   def self.raw_connect(opts)
-    # Create the connection:
-    Connection.new(
-      :host      => opts[:server],
-      :port      => opts[:port],
-      :token     => ManageIQ::Password.try_decrypt(opts[:token]),
-      :namespace => "" # Collect resources across all namespaces
-    )
+    require "kubevirt"
+
+    api_key = ManageIQ::Password.try_decrypt(opts[:token])
+
+    kubevirt_api_client = Kubevirt::ApiClient.new
+    kubevirt_api_client.config.api_key    = api_key
+    kubevirt_api_client.config.scheme     = "https"
+    kubevirt_api_client.config.host       = "#{opts[:server]}:#{opts[:port] || 6443}"
+    kubevirt_api_client.config.verify_ssl = false # TODO check security_protocol
+    kubevirt_api_client.config.logger     = $log # TODO add a kubevirt logger
+    kubevirt_api_client.config.debugging  = Settings.log.level == "debug"
+    kubevirt_api_client.default_headers["Authorization"] = "Bearer #{api_key}"
```

Comment:
Modifying the default headers seems weird to me? Is that just how the interface works?

Reply:
Agree it was weird to me also, it seems like an openapi thing since we do the same in powervs https://github.com/ManageIQ/manageiq-providers-ibm_cloud/blob/master/app/models/manageiq/providers/ibm_cloud/power_virtual_servers/manager_mixin.rb#L27

I think if the openapi spec declared an auth setting then the openapi generator would handle this for you, so far this was the only way to get the openapi generated gem to send authorization headers.